lib/phoenix/endpoint.ex

```diff
@@ -226,6 +226,9 @@ defmodule Phoenix.Endpoint do
       unquote(pubsub())
       unquote(plug())
       unquote(server())
+
+      require Phoenix.Endpoint.Instrument
+      Phoenix.Endpoint.Instrument.definstrument(var!(otp_app))
```

Comment:
I was thinking more about something like this (inside `__before_compile__`):

```elixir
otp_app = Module.get_attribute(env.module, :otp_app)
instrumentation = Phoenix.Endpoint.Instrument.definstrument(otp_app, env.module)

quote do
  unquote(instrumentation)
end
```

There is no reason to use a macro in this particular case. :)